rp2040-hal/src/xosc.rs

```diff
@@ -89,13 +83,12 @@ impl CrystalOscillator<Disabled> {
     }
 
     /// Initializes the XOSC : frequency range is set, startup delay is calculated and set.
-    pub fn initialize(self, frequency: Hertz) -> Result<CrystalOscillator<Initialized>, Error> {
+    pub fn initialize(self, frequency: HertzU32) -> Result<CrystalOscillator<Initialized>, Error> {
         const ALLOWED_FREQUENCY_RANGE: RangeInclusive<Megahertz<u32>> =
-            Megahertz(1)..=Megahertz(15);
-        const STABLE_DELAY: Milliseconds = Milliseconds(1_u32);
-        const DIVIDER: Fraction = Fraction::new(256, 1);
```

Comment:
I feel its somewhat worth it to keep the const even if its just `const DIVIDER: u32 = 256;`